runbotics/runbotics-scheduler/src/scheduler-database/tenant/tenant.service.ts

```diff
@@ -0,0 +1,93 @@
+import { BadRequestException, Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { MoreThanOrEqual, Repository } from 'typeorm';
+
+import { UserEntity } from '#/database/user/user.entity';
+import { Tenant } from './tenant.entity';
+import { CreateTenantDto } from './dto/create-tenant.dto';
+import { UpdateTenantDto } from './dto/update-tenant.dto';
+import { Logger } from '#/utils/logger';
+import { TenantInviteCode } from './tenant-invite-code.entity';
+import dayjs from 'dayjs';
+import { TenantInviteCodeDto } from './dto/invite-code.dto';
+
+@Injectable()
+export class TenantService {
+    private readonly logger = new Logger(TenantService.name);
+
+    constructor(
+        @InjectRepository(Tenant)
+        private readonly tenantRepository: Repository<Tenant>,
+        @InjectRepository(TenantInviteCode)
+        private readonly inviteCodeRepository: Repository<TenantInviteCode>
+    ) {}
+
+    getTenantById(id: string) {
+        return this.tenantRepository.findOneBy({ id });
+    }
+
+    getAllTenants() {
+        return this.tenantRepository.find();
+    }
+
+    async createTenant(tenantDto: CreateTenantDto, requester: UserEntity): Promise<Tenant> {
+        const tenantByName = await this.tenantRepository.findOneBy({ name: tenantDto.name });
+        if (tenantByName) throw new BadRequestException('Name already exist', 'NameExist');
+
+        const newTenant = {
+            ...tenantDto,
+            created: dayjs(),
+            updated: dayjs(),
```

Comment:
If you use Create/UpdateDateColumn in entity the dates will be updated automatically